.github/workflows/public-api.yml

```
@@ -0,0 +1,61 @@
+        trap 'git worktree remove --force "$worktree_dir"' EXIT
+
+        git worktree add --detach "$worktree_dir" "$BASE_SHA"
+
+        python utils/public_api.py extract \
+          --src-root "$worktree_dir/src/aiida" \
+          --output /tmp/public_api_base.json
+
+        if ! python utils/public_api.py diff --exit-code /tmp/public_api_base.json; then
+          cat >&2 <<'EOF'
+
+This check fails whenever the public API changes, so that the change is reviewed explicitly.
```

actionlint (1.7.12)

[error] could not parse as YAML: could not find expected ':'

(syntax-check)